config/initializers/string.rb

```diff
@@ -9,3 +9,9 @@ def secure_compare(other)
     res == 0
   end
 end
+
+class NilClass
+  def secure_compare(_)
+    false
+  end
+end
```

Comment:
I'm not sure I'm comfortable monkey-patching this into `NilClass` everywhere just for one call; is there a reason not to just make the `secure_compare` call in `smoke_testing?` conditional on the `X-Smoke-Test` header existing?